### PR TITLE
Table Panel: Fix images not showing on hover with multiple data links

### PR DESCRIPTION
**Pull request by codeincarnate**
This PR fixes an issue when multiple data links are present with the image cell. When hovered the image won't show and the data links context menu isn't accessible. This fixes this by applying a height to a wrapping link tag, and by removing the usage of the `Button` component which in this case is also unnecessary.

A bit of an elaboration on the related fix here #84625

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by baldm0mma** [comment]
Nice! I removed some unused imports, but my linter's still pretty mad about the use of an anchor instead of a button...
```
Anchor used as a button. Anchors are primarily expected to navigate. Use the button element instead. Learn more: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/anchor-is-valid.mdeslint[jsx-a11y/anchor-is-valid](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/tree/HEAD/docs/rules/anchor-is-valid.md)
Visible, non-interactive elements with click handlers must have at least one keyboard listener.eslint[jsx-a11y/click-events-have-key-events](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/tree/HEAD/docs/rules/click-events-have-key-events.md)
Avoid non-native interactive elements. If using native HTML is not possible, add an appropriate role and support for tabbing, mouse, keyboard, and touch inputs to an interactive content element.eslint[jsx-a11y/no-static-element-interactions](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/tree/HEAD/docs/rules/no-static-element-interactions.md)
```

File: packages/grafana-ui/src/components/Table/ImageCell.tsx [hunk 1]
```diff
@@ -27,13 +23,16 @@ export const ImageCell = (props: TableCellProps) => {
...
+                alt=""
+              />
+            );
             if (api.openMenu) {
-              return (
-                <Button className={cx(clearButtonStyle)} onClick={api.openMenu}>
-                  {img}
-                </Button>
-              );
+              return <a onClick={api.openMenu}>{img}</a>;
```
Comment (nmarrs) [inline packages/grafana-ui/src/components/Table/ImageCell.tsx:35]: To address lint issues I think we can just add:
```suggestion
              return <a onClick={api.openMenu} role="button" tabIndex={0}>{img}</a>;
```

**Review by nmarrs — APPROVED** [review]
LGTM - replaced anchor with div with some a11y props and things seem to work and as a bonus are more accessible :)


https://github.com/grafana/grafana/assets/22381771/4a8346c2-3e08-4ae7-99e1-9a5771c7e0d6

[Test dashboard.json](https://github.com/grafana/grafana/files/15069638/Test.image.hover.disappear-1713831133068.json)

**Review by baldm0mma — APPROVED** [review]
LGTM

**Pull request merged**